discovery.md

```diff
@@ -109,296 +123,217 @@ to execute some logic, which might lead to the occurrence of new events.
-    authscope: [string] ?  (explain)
-    protocols: +
-    - protocol: [string]
-    types: *
-    - type: [ce-type value]
-      description: [human string] ?
-      datacontenttype: [ce-datacontenttype value] ?
-      dataschema: [ce-dataschema URI] ?
-      dataschematype: [string per RFC 2046] ?
-      dataschemacontent: [schema] ?
-      dataschemacontent:
-      extensions: *
-      - name: [CE context attribute name]
-        type: [CE type string]
-        spec: [URI to specification defining the extension] ?
+{
+  "url": "[unique URL to this service]",
+  "description": "[human string]", ?
+  "docsurl": "[URL reference for human documentation]", ?
+  "specversion": "[ce-specversion value]",
+  "subscriptionurl": "[URL to which the Subscribe request will be sent]",
+  "subscriptionconfig": { ?
+    "[key]": "[value]", *
+  },
+  "authscope": "[string]", ?
+  "protocols": [ "[string]" + ],
+  "types": [ ?
+    { *
+      "type": "[ce-type value]",
+      "description": "[human string]", ?
+      "datacontenttype": "[ce-datacontenttype value]", ?
+      "dataschema": "[ce-dataschema URI]", ?
+      "dataschematype": "[string per RFC 2046]", ?
+      "dataschemacontent": "[schema]", ?
+      "extensions": [ ?
+        { *
+          "name": "[CE context attribute name]",
+          "type": "[CE type string]",
+          "specurl": "[URL to specification defining the extension]" ?
+        }
+      ]
+    }
+  ]
+}
 ```
 
 An example:
-```
-- services:
-  - service: example.com
-    subscriptionuri: https://events.example.com
-    protocols:
-    - protocol: HTTP
-    types:
-    - type: com.example.widget.create
-    - type: com.example.widget.delete
+```json
+{
+  "url": "https://example.com/services/widgetService",
+  "specversion": "v1.0",
+  "subscriptionurl": "https://events.example.com",
+  "protocols": [ "HTTP" ],
+  "type": [
+    {
+      "type": "com.example.widget.create",
+      "type": "com.example.widget.delete"
+    }
+  ]
+}
 ```
 
 Note: the above is just a sample and implementations are free to use any
 internal model they wish to store the data as long as they are compliant
 with the wire format/API defined by this specification.
 
-### Entities in the API
-
-1. `Service`
-2. `Type`
-3. `Producer`
+#### Service Attributes
 
-The `Service` and `Type` entities form the basis of the directory and can
-be used to build user experiences around discovery. The `Producer` entity is
-keyed off of `{Service.Name, Type.Name}` and provides the necessary
-details to create a subscription for events of that `ce-type` from the selected
-`Service`.
+The following sections define the attributes that appear in a Service
+entity.
 
-### Entity Specifications
-
-This section details the fields that make up each of the entities referenced
-earlier in this document.
-
-#### `service` entity
-
-Used in discovery for enumerating the different services represented in this
-discovery system.
-
-##### `service` entity attributes
+##### name
```

Comment:
I did not see the `name` in the pseudo json above. Was this intentionally?

Reply:
fixed